### PR TITLE
[ios] Added float equality comparison warning.

### PR DESCRIPTION
**Pull request by igrechuhin**

### CODE REVIEW

**Comment by mapsmetest** [comment]
Jenkins asks: can anyone have a look at this request?
If you want Jenkins to compile and test this, write "test this please" or JT_TP (without the underscore in the middle).

File: iphone/Maps/Classes/Common.h [hunk 1]
```diff
@@ -89,3 +89,9 @@ static inline NSString * bcp47ToTwineLanguage(NSString const * bcp47LangName)
   // Taking two first symbols of a language name. For example ru-RU -> ru
   return [bcp47LangName substringToIndex:2];
 }
+
+// Use only for screen dimensions CGFloat comparison
+static inline BOOL fequal(CGFloat left, CGFloat right)
+{
+  return fabs(left - right) < 0.5;
+}
```
Comment (VladiMihaylenko) [inline iphone/Maps/Classes/Common.h]: нужен перенос строки

Comment (ygorshenin) [inline iphone/Maps/Classes/Common.h]: Мб отразить факт что эта функция будет использоваться только для сравнения размерностей в ее названии? equalScreenDimensions?

Comment (VladiMihaylenko) [inline iphone/Maps/Classes/Common.h]: имхо, сложно :)

**Comment by VladiMihaylenko** [comment]
LGTM

**Comment by igrechuhin** [comment]
@deathbaba PTAL & merge

File: iphone/Maps/Classes/Common.h (hunk 1 above)
Comment (ygorshenin) [inline iphone/Maps/Classes/Common.h]: Ке? Тогда велик шанс что fequal будет переиспользован и для обычных чисел с плавающей точкой, для которых точность в 0.5 неприемлима. Что сложно? find-and-replace? :)

**Comment by ygorshenin** [comment]
LGTM

File: iphone/Maps/Classes/CustomViews/MapViewControls/ZoomButtons/MWMZoomButtonsView.mm [hunk 2]
```diff
@@ -118,7 +119,8 @@ - (CGFloat)topBound
 
 - (void)setBottomBound:(CGFloat)bottomBound
 {
-  if (_bottomBound == bottomBound)
+
```
Comment (deathbaba) [inline iphone/Maps/Classes/CustomViews/MapViewControls/ZoomButtons/MWMZoomButtonsView.mm]: extra \n

**Comment by deathbaba** [comment]
Are there real bugs because of float comparison, or a "let's add extra warning key and fix everything, even if it's not necessarily needed" decision? It can lead to errors only if floats come from different sources/after math computations. If you always get screen bounds from system/resources it should be safe to compare floats for equality.
You can also consider using base/math.hpp AlmostEqualULPs() which was designed for efficient floats comparison.
LGTM if it is really needed to fix a bug, and it is better to remove -Wfloat-equal otherwise.

**Comment by igrechuhin** [comment]
In UI due to animations we can get fractional values, however the only valuable change is the one greater than 0.5
Excess UI updated are not required and are unwanted.
Apple recommends UI coordinates to be without fractional part.
I believe we should enable this warning.

**Comment by deathbaba** [comment]
If there are animation computations - I agree, it should be enabled on iOS code level.

**Pull request merged**